tests/integration/cattletest/core/common_fixtures.py

```diff
@@ -231,6 +231,24 @@ def wait_transitioning(client, obj, timeout=DEFAULT_TIMEOUT):
     return client.wait_transitioning(obj, timeout=timeout)
 
 
+@pytest.fixture
+def wait_until_expected_state(client, resource, expected_state,
```

Comment:
I guess I ran into this when there were multiple state changes.  So the assert would occasionally fail depending on how quickly the resource went through the transitions.